Source/RealmThreadHelper.swift

```diff
@@ -9,23 +9,38 @@
 import Foundation
 
 
-/**
-Execute a block in the main thread. 
-If we are already in it, just execute. If not, do a dispatch to execute it.
-
-You can select if the execution should be sync or async (careful with deadlocks!!)
-
-- parameter sync:  Bool, true if the execution should be dispatch_sync, false for dispatch_async
-- parameter block: Block to execute
-*/
-func executeOnMainThread(sync: Bool = false, block: ()->()) {
-    if NSThread.currentThread().isMainThread {
-        block()
-    }
-    else if sync {
-        dispatch_sync(dispatch_get_main_queue(), block)
+struct Threading {
+    static let isTesting: Bool = {
+        let environment = NSProcessInfo.processInfo().environment
+        return environment["TEST"] != nil
+    }()
+    
+    /**
+     Execute a block in the main thread.
+     If we are already in it, just execute. If not, do a dispatch to execute it.
+     
+     You can select if the execution should be sync or async (careful with deadlocks!!)
+     
+     - parameter sync:  Bool, true if the execution should be dispatch_sync, false for dispatch_async
+     - parameter block: Block to execute
+     */
+    static func executeOnMainThread(sync: Bool = false, block: ()->()) {
+        guard !NSThread.currentThread().isMainThread else {
+            block()
+            return
+        }
+        executeOnQueue(dispatch_get_main_queue(), sync: sync, block: block)
     }
-    else {
-        dispatch_async(dispatch_get_main_queue(), block)
+    
+    /**
+     Execute a block in the passed queue.
+     
+     You can select if the execution should be sync or async (careful with deadlocks!!)
+     
+     - parameter sync:  Bool, true if the execution should be dispatch_sync, false for dispatch_async
+     - parameter block: Block to execute
+     */
+    static func executeOnQueue(queue: dispatch_queue_t, sync: Bool = false, block: ()->()) {
+        sync && !isTesting ? dispatch_sync(queue, block) : dispatch_async(queue, block)
```

Comment:
Would be more readable to separate isTesting logic. What happens if isTesting is true? Do you care about the "sync" param? 
```
if isTesting { whatever and return }
sync ? blabla : bloblo
```